mk/spksrc.cross-rust.mk

```diff
@@ -31,93 +31,38 @@ TC = syno$(ARCH_SUFFIX)
 endif
 
 ##### rust specific configurations
+include ../../mk/spksrc.cross-rust-env.mk
 
 # configure is used to install rust targets
-CONFIGURE_TARGET = install_rust_target
+CONFIGURE_TARGET = nop
 
-# skip compile_target if not used by module
 ifeq ($(strip $(COMPILE_TARGET)),)
-COMPILE_TARGET = nop
+COMPILE_TARGET = rust_build_target
 endif
 
-RUST_TOOLCHAIN ?= stable
-
-RUST_TARGET =
-# map archs to rust targets
-ifeq ($(findstring $(ARCH), $(x64_ARCHS)),$(ARCH))
-RUST_TARGET=x86_64-unknown-linux-gnu
-endif
-ifeq ($(findstring $(ARCH), $(i686_ARCHS)),$(ARCH))
-RUST_TARGET=i686-unknown-linux-gnu
-endif
-ifeq ($(findstring $(ARCH), $(ARMv5_ARCHS)),$(ARCH))
-# may be not supported for cargo
-RUST_TARGET=armv5te-unknown-linux-gnueabi
-endif
-ifeq ($(findstring $(ARCH), $(ARMv7_ARCHS)),$(ARCH))
-RUST_TARGET=armv7-unknown-linux-gnueabihf
-endif
-ifeq ($(findstring $(ARCH), $(ARMv7L_ARCHS)),$(ARCH))
-RUST_TARGET=armv7-unknown-linux-gnueabi
-endif
-ifeq ($(findstring $(ARCH), $(ARMv8_ARCHS)),$(ARCH))
-RUST_TARGET=aarch64-unknown-linux-gnu
-endif
-ifeq ($(findstring $(ARCH), $(PPC_ARCHS)),$(ARCH))
-RUST_TARGET=powerpc-unknown-linux-gnu
-endif
-ifeq ($(RUST_TARGET),)
-$(error Arch $(ARCH) not supported)
-endif
-
-# Use distrib folder as cargo download cache
-CARGO_HOME_PATH=/spksrc/distrib/cargo
-ENV += CARGO_HOME=$(CARGO_HOME_PATH)
-ENV += PATH=:$(CARGO_HOME_PATH)/bin/:$(PATH)
```

Comment:
This is the contents of $CARGO_HOME/env:
```
#!/bin/sh
# rustup shell setup
# affix colons on either side of $PATH to simplify matching
case ":${PATH}:" in
    *:"$HOME/.cargo/bin":*)
        ;;
    *)
        # Prepending path in case a system-installed rustc needs to be overridden
        export PATH="$HOME/.cargo/bin:$PATH"
        ;;
esac

```

Adding the bin $CARGO_HOME/bin dir to the path is necessary

Reply:
Indeed but as the `CARGO_HOME` is being set to `/usr/local/` therefore `/usr/local/bin` is automagically being in the default `PATH`.

Reply:
I figured using cargo_home in a dir that has other files would conflict. There's a lib and cache dir and some other stuff in there too